tracer/src/Datadog.Trace.Tools.dd_dotnet/Checks/ProcessBasicCheck.cs

```diff
@@ -643,53 +694,22 @@ private static bool GetLocalMachineSubKeyVersion(string uninstallKey, string dat
             return versionFound;
         }
 
-        internal static void CheckLinuxInstallation(string installDirectory)
+        private static bool CheckEnableDiagnostics(ProcessInfo process)
         {
-            string archFolder;
-            var osArchitecture = RuntimeInformation.OSArchitecture;
+            var variablesToCheck = new[] { "DOTNET_EnableDiagnostics", "COMPlus_EnableDiagnostics", "DOTNET_EnableDiagnostics_Profiler", "COMPlus_EnableDiagnostics_Profiler" };
 
-            if (osArchitecture == Architecture.X64)
-            {
-                archFolder = Utils.IsAlpine() ? "linux-musl-x64" : "linux-x64";
-            }
-            else if (osArchitecture == Architecture.Arm64)
-            {
-                archFolder = "linux-arm64";
-            }
-            else
-            {
-                Utils.WriteError(UnsupportedLinuxArchitecture(osArchitecture.ToString()));
-                return;
-            }
+            bool result = true;
 
-            try
+            foreach (var key in variablesToCheck)
```

Comment:
Purely out of interest, can we use `[]` if we use it directly in a foreach?

```suggestion
            foreach (var key in ["DOTNET_EnableDiagnostics", "COMPlus_EnableDiagnostics", "DOTNET_EnableDiagnostics_Profiler", "COMPlus_EnableDiagnostics_Profiler"])
```

Reply:
Since it uses target-typing, nope. Doesn't compile :(